packages/react-server/src/ReactFlightServer.js

```diff
@@ -437,6 +518,33 @@ export function resolveModelToJSON(
     return null;
   }
 
+  if (
+    value &&
+    value.$$typeof === REACT_SERVER_CONTEXT_TYPE &&
+    key === '__pop'
+  ) {
+    popProvider((value: any));
+    if (__DEV__) {
+      insideContextProps = null;
+      isInsideContextValue = false;
+    }
+    return (undefined: any);
+  }
+
+  if (value.$$typeof === REACT_PROVIDER_TYPE) {
```

Comment:
You check for `value` being truthy in the other `if` above this but not here. Either it's never null/undefined or you need the check here too. Can it be undefined?

I'd just solve this by moving them into the `typeof value === 'object'` branch below though so that you only check this on actual objects. They're never null because there's a check above.

Reply:
You can also make it a switch statement.

Since you probably match fewer contexts than module references in a tree (given that each child component is a module reference) you can probably move this after the isModuleReference check.